x-pack/plugins/ingest_manager/server/services/epm/fields/field.test.ts

```diff
@@ -179,4 +179,35 @@ describe('processFields', () => {
       JSON.stringify(mixedFieldsExpanded)
     );
   });
+
+  const objectFieldWithProperty = [
```

Comment:
Will it also work if you have `a.b` first and then `a`?

Reply:
Yes. I've added a unit test for that.